### ISSUE TITLE
comparison with openpiv-python 

### ISSUE THREAD

**Issue by alexlib** [ali-sh-96/openpiv-python-cpu#1]
Dear @ali-sh-96 - I'd like to get some of your experience with openpiv-python-cpu comparison with openpiv-python. I've reviewed the code; it's very well-written and exciting. However, I'd like to understand better what terms it's refined: speed/uncertainty/new features. 

I think we can discuss the option of incorporating parts of the new code into the original code or maybe replacing the original library completely —if we manage to keep some backward compatibility with many users who might want to use it differently.

**Comment by ali-sh-96** [comment]
Hi Prof. Liberzon,
Thanks for taking the time to review the code. One objective for developing the new CPU version was to get a version that is compatible with openpiv-python-gpu, written by Eric Yang. I tried my best to write the code using high-level objects, same as those in the GPU code. The new version is optimized for speed, specially since it uses FFTW for cross correlation not numpy. There are some new features. For example being able to use single precision for PIV, which significantly increases the speed. However, it also doesn't have some of the features of the older CPU version. For example, it only accepts square windows (no rectangular windows) that are powers of 2. This is for performance reasons. I have demonstrated the accuracy in the paper on researchgate.

**Comment by ali-sh-96** [comment]
It is possible to incorporate most of the new algorithms in the older version, such as validation and outlier replacement. Part of the code is also the same as before, with minor changes. But the cross correlation and passes are very  different.

**Comment by alexlib** [comment]
Could we please video chat to understand the main differences and brainstorm the merge of the two software ? 
alex liberzon at gmail dot com

**Comment by ali-sh-96** [comment]
Yes that's fine with me. I'm in Toronto right now, est time zone. Do you prefer email for communications? Please let me know your availabilities for the meeting.

**Comment by alexlib** [comment]
yes please let's move to email. 
your morning time and my afternoon are fine from my side
